### ISSUE TITLE
crash processing m3u playlist

### ISSUE THREAD

**Issue by hmoffatt** [gerbera/gerbera#2463]
I just updated to 1.10 from 1.9 and started with a fresh database. The container exits after processing an m3u playlist. If I start it again, the same happens with the next playlist.

```
$ docker-compose up
Creating gerbera ... done
Attaching to gerbera
gerbera    | 2022-02-26 08:41:56   info: Gerbera UPnP Server 1.10.0 - http://gerbera.io/
gerbera    | 2022-02-26 08:41:56   info: Copyright 2016-2021 Gerbera Contributors.
gerbera    | 2022-02-26 08:41:56   info: Gerbera is free software, covered by the GNU General Public License version 2
gerbera    | 2022-02-26 08:41:56   info: Loading configuration from: /var/run/gerbera/config.xml
gerbera    | 2022-02-26 08:41:56   info: Checking configuration...
gerbera    | 2022-02-26 08:41:56   info: Configuration check succeeded.
gerbera    | 2022-02-26 08:41:56   info: Loading 0 configuration items from database
gerbera    | 2022-02-26 08:41:56   info: Loading autoscan location: /content; recursive: true, last_modified: unset
gerbera    | 2022-02-26 08:41:56   info: Initialising UPnP with interface: <unset>, port: 49494
gerbera    | 2022-02-26 08:41:56   info: IPv4: Server bound to: 192.168.42.18:49494
gerbera    | 2022-02-26 08:41:56   info: IPv6: Server bound to: fe80::3ac9:86ff:fe40:8637:49494
gerbera    | 2022-02-26 08:41:56   info: IPv6 ULA/GLA: Server bound to: <redacted>:49494
gerbera    | 2022-02-26 08:41:56   info: Will send UPnP Alive advertisements every 870 seconds
gerbera    | 2022-02-26 08:41:56   info: The Web UI can be reached by following this link: http://192.168.42.18:49494
gerbera    | 2022-02-26 08:41:56   info: Processing playlist: /content/My One And Only Thrill/Melody Gardot - My One And Only Thrill.m3u
gerbera exited with code 139
```

Or

```
gerbera    | 2022-02-26 08:46:17   info: Processing playlist: /content/Jazz/Bob Mintzer - Live At MCG/Bob Mintzer - Live At MCG.m3u
```

or

```
gerbera    | 2022-02-26 08:46:53   info: Processing playlist: /content/Jazz/Vince Jones and Grace Knight - Come in spinner/Vince Jones and Grace Knight - Come in spinner.m3u
```

There's nothing special about those m3u files that I see.

**Comment by ddg1** [comment]
Not sure if it's the same issue, but something similar happened to me. My custom script was failing because `readln()` would only return an empty string. I tried it with the default scripts and a standard m3u file and got the same behavior.

I've looked into it and `fgets` never returns anything in my case: https://github.com/gerbera/gerbera/blob/be8273cf8528a33f11f263bb64419356c9d19a54/src/content/scripting/playlist_parser_script.cc#L200-L201

If I change the way the file handle is opened in https://github.com/gerbera/gerbera/blob/be8273cf8528a33f11f263bb64419356c9d19a54/src/content/scripting/playlist_parser_script.cc#L222-L223 to `currentHandle = std::fopen(obj->getLocation().c_str(), "re");`, my scripts seem to run fine and playlists get loaded correctly.

I think what's happening is that the `GrbFile` is being destroyed, so the file gets closed before it is read, but this is mostly speculation on my part.
